### PR TITLE
Attach sequence number, publisher GID, and source timestamp to public…

### PR DESCRIPTION
**Pull request by clalancette**
…ations.

That way, the subscriptions can pull them out of the attachment and pass it to the upper layers.

### CODE REVIEW

File: rmw_zenoh_cpp/src/rmw_zenoh.cpp [hunk 1]
```diff
@@ -775,6 +776,48 @@ rmw_return_loaned_message_from_publisher(
   return RMW_RET_UNSUPPORTED;
 }
 
+static z_owned_bytes_map_t create_map_and_set_sequence_num(
```
Comment (Yadunund) [inline rmw_zenoh_cpp/src/rmw_zenoh.cpp:779]: Should we include this in an anonymous namespace?

File: rmw_zenoh_cpp/src/detail/rmw_data_types.cpp [hunk 2]
```diff
@@ -232,6 +247,7 @@ std::unique_ptr<ZenohReply> rmw_client_data_t::pop_next_reply()
 }
 
 //==============================================================================
+
```
Comment (Yadunund) [inline rmw_zenoh_cpp/src/detail/rmw_data_types.cpp]: Nit: Unintended newline?

**Review by Yadunund — APPROVED** [review]
LGTM with minor comments to optionally address. The changes here will allow us to very easily detect `MESSAGE_LOST` events!

File: rmw_zenoh_cpp/src/detail/rmw_data_types.cpp (hunk 2 above)
Comment (clalancette) [reply rmw_zenoh_cpp/src/detail/rmw_data_types.cpp]: Oops, good call.  Will fix that one.

File: rmw_zenoh_cpp/src/rmw_zenoh.cpp (hunk 1 above)
Comment (clalancette) [reply rmw_zenoh_cpp/src/rmw_zenoh.cpp:779]: We aren't consistent here.  Sometimes we use the anonymous namespace, sometimes we use `static`.  They both accomplish the same thing.  I personally find the `static` to be easier to read (as you don't have to look at the namespace to find out whether it is a public function or not), but I don't feel strongly about it.

For now, I'm going to leave this as-is.  I'll open a separate issue to make this consistent throughout the codebase.

File: rmw_zenoh_cpp/src/detail/rmw_data_types.cpp (hunk 2 above)
Comment (clalancette) [reply rmw_zenoh_cpp/src/detail/rmw_data_types.cpp]: Fixed in fa6f257.

**Pull request merged**